### PR TITLE
feat: configure eol character for git.commitEntry

### LINKED ISSUE #679: `git.commitEntry` make end of line style configurable
### Clear and concise description of the problem

Current implementation sets the EOL style to `'\r\n'`. 

### Suggested solution

Add an option to configure the EOL character to the users preferred style
```typescript
commitEntry(options: { 
  merge?: boolean,
  eol?: 'LF' | 'CRLF' 
} = {}): string {
    let eolChar = '\r\n';
    if (options.eol === 'LF') {
        eolChar = '\n';
    }

    let entry = 'commit {{git.commitSha}}' + eolChar;

    if (options.merge || this.faker.datatype.number({ min: 0, max: 4 }) === 0) {
      entry += 'Merge: {{git.shortSha}} {{git.shortSha}}' + eolChar;
    }

    entry +=
      'Author: {{name.firstName}} {{name.lastName}} <{{internet.email}}>' + eolChar;
    entry += 'Date: ' + this.faker.date.recent().toString() + eolChar;
    entry +=  eolChar + 'xa0\xa0\xa0\xa0{{git.commitMessage}}' + eolChar;

    return this.faker.fake(entry);
 }
```

### Alternative

_No response_

### Additional context

Use `'\r\n'` as default EOL character to not break current implementations.

### PR DESCRIPTION
**Pull request by xDivisionByZerox**
Fixes #679

### CODE REVIEW

**Review by import-brain — DISMISSED** [review]

File: src/git.ts [hunk 1]
```diff
@@ -50,6 +50,9 @@ export class Git {
    *
    * @param options Options for the commit entry.
    * @param options.merge Set to `true` to generate a merge message line.
+   * @param options.eol Choose the end of line character to use. Defaults to 'CRLF'.
+   * 'LF' = '\n',
+   * 'CRLF' = '\r\n'
```
[the remark this answers is not in the document]
Comment (import-brain) [reply src/git.ts:55]: Yeah, I would want to keep that in too

**Review by ST-DDT — APPROVED** [review]

**Review by import-brain — APPROVED** [review]

**Review by Shinigami92 — CHANGES_REQUESTED** [review]
We need a test for the parameter

File: src/git.ts [hunk 2]
```diff
@@ -59,17 +62,29 @@ export class Git {
...
+    lines.push(
+      `Author: ${this.faker.name.firstName()} ${this.faker.name.lastName()} <${this.faker.internet.email()}>`,
+      `Date: ${this.faker.date.recent().toString()}`,
+      '',
+      `\xa0\xa0\xa0\xa0${this.commitMessage()}`,
+      // to end with a eol char
+      ''
+    );
+
+    const eolChar = options.eol === 'CRLF' ? '\r\n' : '\n';
```
Comment (ST-DDT) [inline src/git.ts:86]: This condition doesn't look like CRLF is the default.

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: src/git.ts (hunk 2 above)
Comment (xDivisionByZerox) [reply src/git.ts]: Yeah, your right. Was noticing this when writing the test^^

File: test/git.spec.ts [hunk 3]
```diff
@@ -141,6 +141,34 @@ describe('git', () => {
             expect(parts[4]).toMatch(/^\s{4}.+$/);
           }
         });
+
+        it('should return a random commitEntry with a default end of line charcter of "\n\r"', () => {
```
Comment (ST-DDT) [inline test/git.spec.ts:145]:
```suggestion
        it('should return a random commitEntry with a default end of line charcter of "\r\n"', () => {
```

File: test/git.spec.ts [hunk 4]
```diff
@@ -141,6 +141,34 @@ describe('git', () => {
...
+          const commitEntry = faker.git.commitEntry();
+          const parts = commitEntry.split('\r\n');
+
+          expect(parts.length).toBeGreaterThanOrEqual(6);
+          expect(parts.length).toBeLessThanOrEqual(7);
+        });
+
+        it('should return a random commitEntry with an explicit end of line charcter of "\n\r" with eol = CRLF', () => {
```
Comment (ST-DDT) [inline test/git.spec.ts:153]:
```suggestion
        it('should return a random commitEntry with a configured end of line charcter of "\r\n" with eol = CRLF', () => {
```

File: test/git.spec.ts [hunk 5]
```diff
@@ -141,6 +141,34 @@ describe('git', () => {
...
+          const commitEntry = faker.git.commitEntry({
+            eol: 'CRLF',
+          });
+          const parts = commitEntry.split('\r\n');
+
+          expect(parts.length).toBeGreaterThanOrEqual(6);
+          expect(parts.length).toBeLessThanOrEqual(7);
+        });
+
+        it('should return a random commitEntry with a configured end of line charcter of "\n" with eol = LF', () => {
```
Comment (ST-DDT) [inline test/git.spec.ts:163]: This test would still pass with eol crlf. 
Maybe add a expect.not.contains(crlf)?

**Review by ST-DDT — CHANGES_REQUESTED** [review]

File: test/git.spec.ts (hunk 5 above)
Comment (xDivisionByZerox) [reply test/git.spec.ts:163]: done in 0e04960

File: test/git.spec.ts (hunk 4 above)
Comment (xDivisionByZerox) [reply test/git.spec.ts]: done in 7130cdbf2034dc75fcce451748ecfa0001d81c12

File: test/git.spec.ts (hunk 3 above)
Comment (xDivisionByZerox) [reply test/git.spec.ts]: done in 7130cdbf2034dc75fcce451748ecfa0001d81c12

**Review by ST-DDT — APPROVED** [review]

**Review by Shinigami92 — APPROVED** [review]

**Pull request merged**